lib/service/cfg.go

```diff
@@ -541,6 +542,38 @@ type App struct {
 	Rewrite *Rewrite
 }
 
+// Check validates an application.
+func (a App) Check() error {
+	if a.Name == "" {
+		return trace.BadParameter("missing name")
```

Comment:
I'd add context to error messages. If the message is output without context, it is pretty hard to understand where the problem is. Something like `missing application name` can go a long way.